### ISSUE TITLE
Install fails on macOS: proxy_env[@] unbound variable

### ISSUE THREAD

**Issue by hoangnb24** [Dicklesworthstone/beads_rust#22]
Platform: macOS (darwin_arm64)
Version: v0.1.7

**Error**: Installer exits with 'proxy_env[@]: unbound variable' and br is not detected.

> bash: line 923: proxy_env[@]: unbound variable
> [ERR ] Beads Rust installer finished but 'br' was not detected. Ensure its install directory is on PATH or rerun with --skip-beads to handle installation manually.

**Expected**: Installer completes without errors and br is available on PATH.

**Comment by Dicklesworthstone** [comment]
Fixed. Changed the problematic lines in `install.sh`:

```bash
# Before (fails with set -u when proxy_env is empty):
env "${proxy_env[@]}" curl ...

# After (safe expansion):
env ${proxy_env[@]+"${proxy_env[@]}"} curl ...
```

The `${arr[@]+"${arr[@]}"}` pattern expands to nothing when the array is unset/empty, avoiding the "unbound variable" error under `set -u`.

**Comment by Dicklesworthstone** [comment]
**Final Resolution** (commit 64d86fb)

Fixed both curl and wget invocations in `install.sh` using the standard bash idiom for empty array handling:

```bash
env ${proxy_env[@]+"${proxy_env[@]}"} curl ...
env ${proxy_env[@]+"${proxy_env[@]}"} wget ...
```

This pattern:
- Expands to `"${proxy_env[@]}"` when the array is set (even if empty in bash 4.4+)
- Expands to nothing when the array is unset
- Works correctly under `set -u` on all bash versions including macOS default bash
